client/components/allProducts.js

```diff
@@ -0,0 +1,56 @@
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import {ProductListing} from './productListing'
+
+class allProducts extends Component {
+  //CHECK ASSUMPTION: FETCHPRODUCTS IN MAIN'S COMPONENT DID MOUNT?
```

Comment:
This will depend on what the main page ends up being, right? I think that the products should only be fetched on the page they need to be listed, so if `<AllProducts />` ends up being the main page, then yes.